kube-controllers/pkg/controllers/loadbalancer/loadbalancer_controller.go

```diff
@@ -369,38 +381,19 @@ func (c *loadBalancerController) syncIPAM() {
 
 	for _, svc := range services {
 		if IsCalicoManagedLoadBalancer(svc, c.cfg.AssignIPs) {
-			handle, err := createHandle(svc)
+			svcObj, err := serviceObjectFromService(svc)
 			if err != nil {
-				break
-			}
-			svcObj := serviceObject{
-				handle: handle,
-				namespacedName: types.NamespacedName{
-					Namespace: svc.Namespace,
-					Name:      svc.Name,
-				},
+				log.WithError(err).Error("Error getting service object from service")
+				continue
 			}
-			svcObjs[svcObj] = true
+			svcObjs[*svcObj] = true
 		}
 	}
 
-	for _, block := range c.ipamBlocks {
-		b := block.Value.(*model.AllocationBlock)
-		for i := range b.Allocations {
-			if b.Allocations[i] != nil {
-				namespacedName := types.NamespacedName{
-					Namespace: b.Attributes[*b.Allocations[i]].AttrSecondary[ipam.AttributeNamespace],
-					Name:      b.Attributes[*b.Allocations[i]].AttrSecondary[ipam.AttributeService],
-				}
-				handle := b.Attributes[*b.Allocations[i]].AttrPrimary
-
-				svcObj := serviceObject{
-					handle:         *handle,
-					namespacedName: namespacedName,
-				}
-
-				svcObjs[svcObj] = true
-			}
+	for block := range c.allocationTracker.ipsByBlock {
+		for ip := range c.allocationTracker.ipsByBlock[block] {
+			svcObj := c.allocationTracker.servicesByIP[ip]
+			svcObjs[svcObj] = true
```

Comment:
```suggestion
	for svcObj := range c.allocationTracker.ipsByService {
		svcObjs[svcObj] = true
```

I think we'll have already condensed the IPAM database into a map of service objects that we can iterate quicker, right? Or is there a reason we want to iterate all of the IPs here?